pa4/checker-pa4.ml

```diff
@@ -861,6 +861,19 @@ let main () = begin
 						exit 1;
 					| _ -> failwith "cannot happen: Unknown Method Redefines Error";
 				) supclasses;
+
+
+				(* Type check method expression *)
+				let init_type = exp_typecheck o_e m_e exp in
+	  			let (return_loc, return_type) = ret_type in
+	  			if is_subtype init_type (Class return_type) then
```

Comment:
we can rewrite it so that to get rid of 'then' branch.
If we replace if condition to `not (is_subtype init_type (Class return_type))`

But it is not critical

Reply:
One could make changes in the next commit